### DISCUSSION TITLE
Ping via Custom Script

### DISCUSSION THREAD

**Discussion by but-i-am-dominator** [netbox-community/netbox#5888] [Q&A]
Hello Team, 
Hopefully this is an easy question but, I just can't figure it out. I created a custom script to support a business process. Part of the process is identifying new hosts on the network with ping. My script works great when I run it externally. After modifying it according to your documentation, the ping functionality doesn't work. (I tried several different methods and no dice.) I believe it has something to do with how the script is being called. 

Any help would be appreciated!

**Comment by jeremystretch** [comment]
Can you share what code you have so far? Is it just the ping transmission that's failing, or the entire script?

**Reply by but-i-am-dominator** [reply]
My script is kind of large but, this is the test that I can't get working. 
So far, I have tried different versions of the subprocess and os modules. I also adjusted the WSGI timeout value.

```
import subprocess
from extras.scripts import *

class MyScript(Script):
    class Meta:
        name = "Script Test"

    def run(self, data):
        result = subprocess.run(["/usr/bin/ping", "-c", "1", "127.0.0.1"])
        self.log_success(result)
        self.log_success("Cool Runs")
   ```
   
   Thanks so much for the super fast response!

**Reply by but-i-am-dominator** [reply]
When I run that subprocess call in netbox's virtual environment, there is no issues. 

 When running different calls as a custom script:

result = subprocess.run(["/usr/bin/ping", "-c", "1", "127.0.0.1"])    #### loops forever
result = subprocess.check_output(["/usr/bin/ping", "-c", "1", "127.0.0.1"]) #### raises a CalledProcessError

**Reply by jeremystretch** [reply]
I wonder if it's a permissions issue. Try running it directly in python with `sudo -u netbox python` (or replace `netbox` with whatever user account NetBox runs as).

**Reply by but-i-am-dominator** [reply]
no issues with my configured user running that script (I do use netbox).

**Reply by hSaria** [reply]
Not sure if you still need help, but you might want to try invoking a shell for the command
```python
# Notice that when using `shell=True`, the command is not a list
subprocess.run("ping -c 1 127.0.0.1", shell=True)
```

Also, you'll want to pipe your output if you want to read it
```python
# You can do the same for `stderr`
>>> s = subprocess.run(["ping", "-c", "1", "127.0.0.1"], stdout=subprocess.PIPE)
>>> print(s.stdout.decode())
PING 127.0.0.1 (127.0.0.1): 56 data bytes
64 bytes from 127.0.0.1: icmp_seq=0 ttl=64 time=0.054 ms

--- 127.0.0.1 ping statistics ---
1 packets transmitted, 1 packets received, 0.0% packet loss
round-trip min/avg/max/stddev = 0.054/0.054/0.054/0.000 ms
```
